### PR TITLE
Add some utilities to local testing

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
Adding two utilities:
 - setting logging level for arbitrary target
 - checking hash of a finalized block (given its number)

which I found really useful while working on AZ-395. Hopefully, they will be helpful for someone else as well.

### CODE REVIEW

File: local-tests/chainrunner/node.py
Comment (h4nsu) [inline local-tests/chainrunner/node.py:27]: Couldn't you just add the logging info to `flags`? Seems to me like now there are two incompatible ways of doing the same thing and no logic to disambiguate in case of conflict...

File: local-tests/chainrunner/node.py
Comment (h4nsu) [inline local-tests/chainrunner/node.py:74]: nit: I'd suggest a bit different naming here. This doesn't check anything, it's a getter. How about `get_hash_of` or (even better) `get_hash`? Also the argument would be more self-explanatory if it was called `height`

**Comment by maciejzelaszczyk** [comment]
I see no obvious problems. As I am not yet familiar with our codebase, I will defer from approving at this time and wait for a parallel review by @h4nsu. On a side note, type annotations could help us get rid of the type information from the docstrings.

**Comment by pmikolajczyk41** [comment]
> On a side note, type annotations could help us get rid of the type information from the docstrings.

Glad to hear that, I am also supporting type hints. However, the author of this part of code, @h4nsu, is on the dark side of the force :smile:

File: local-tests/chainrunner/node.py
Comment (h4nsu) [inline local-tests/chainrunner/node.py:76]: A small problem I see with this is that the entry you grep for is logged on a debug level. Otoh there's no info-level message that you can use to get the full hash (only something like `0x0cc8…e968`)

**Review by h4nsu — CHANGES_REQUESTED** [review]

File: local-tests/chainrunner/node.py
Comment (pmikolajczyk41) [reply local-tests/chainrunner/node.py:27]: well, the debug flags have the form of `-ltarget=level`, so it neither matches neither value nor non-value flags that we are using

File: local-tests/chainrunner/node.py
Comment (pmikolajczyk41) [reply local-tests/chainrunner/node.py:76]: You are right, so what about one of the two solutions:
 - adding to docstring a warning that debug level should be on
 - checking in code whether a debug level is on, and if not, printing warning
 ?

File: local-tests/chainrunner/node.py
Comment (pmikolajczyk41) [reply local-tests/chainrunner/node.py]: done, thx

File: local-tests/chainrunner/node.py
Comment (h4nsu) [reply local-tests/chainrunner/node.py]: Could be done with the long form `--log target=level`

**Review by kostekIV — APPROVED** [review]

**Review by h4nsu — APPROVED** [review]

**Review by maciejzelaszczyk — APPROVED** [review]

**Pull request merged**
